3-extra/1-count-words.js

```diff
@@ -25,8 +25,30 @@
 
 function countWords(string) {
   const wordCount = {};
+  let numOfWords,wordPos,theWord;
 
-  // write code here
+  let listOfWords = string.split(' ');
+  /*
+     When a string is empty, split returns [ '' ]
+     Cater for it
+  */
+
+  if (!listOfWords[0].length) // just in case there are no words?
```

Comment:
if you want to check that there are no words, then you would need to do ``!listOfWords.length`` instead. However, this is not needed because the ``while`` loop in line 39 won't be executed when the length is 0